### DISCUSSION TITLE
Allow to see event data for each event

### DISCUSSION THREAD

**Discussion by bigbasti** [umami-software/umami#2845] [Ideas]
Ok the title is a bit quirky, but it is really simple:

I use the event javascript api to track user events on the site, so i send events like:

`umami.track("my_event", {data1: "value1", data2: "value2", ... }`

On the UI i can see all the events and all the data but i can not see the event as a whole.
I was expecting to see the same event on the umami UI, to see that "my_event" was called and that the event had the data `{data1: "value1", data2: "value2", ... }` assosiated with it.

Is there a way to see that or is it maybe planned for v3?

Thanks

**Comment by mystiker** [comment]
Yes please - that would be very helpful. 

I have some troubles with the 50 character limit for event names because I use overly complex names like: "Home-SliderX-ButtonY-Clicked" just because its very convenient to see this information in the session details page.

However I don't think this is what the devs intended and I would rather send "Home-Whatever" with data like {slider: "x", button: "y"} but its not possible to see this information in relevant places, like the mentioned session details page.

**Comment by MajorTom-LunarAlpha** [comment]
Yeah, this is definitely something that’s really missing, in my opinion.

I really want to quickly see event data on the session page when looking at a specific user. This would be gold!

**Comment by rijkdw** [comment]
+1 this would be great.  I'd personally even be satisfied with only being able to export the raw individual event data, to analyze myself later.

**Comment by vsly-ar** [comment]
+1 deal-breaker 
Btw, is this data currently stored or thrown away? 
Association of the concrete event and its data. Because you can only view summarized event data in a pie chart.
